src/profile/webid.test.ts

```diff
@@ -146,22 +157,62 @@ describe("getProfileAll", () => {
     };
 
     mockedFetcher.fetch.mockResolvedValueOnce(
-      new Response(undefined, { headers: { "Content-Type": "text/turtle" } })
+      new Response(await triplesToTurtle(toRdfJsQuads(MOCK_PROFILE)), {
+        headers: {
+          "Content-Type": "text/turtle",
+        },
+        url: "https://some.profile",
+      } as ResponseInit)
     );
+
+    const profileContent = buildThing({ url: "https://some.profile" })
+      .addIri(foaf.primaryTopic, MOCK_WEBID)
+      .build();
+
+    const webIdProfile = setThing(
+      mockSolidDatasetFrom(MOCK_WEBID),
+      profileContent
+    );
+    // Mock the webid unauthenticated fetch
+    const mockedUnauthFetch = jest.requireMock("cross-fetch") as {
+      fetch: jest.Mock<
+        ReturnType<typeof window.fetch>,
+        [RequestInfo, RequestInit?]
+      >;
+    };
+    mockedUnauthFetch.fetch.mockResolvedValueOnce(
+      new Response(await triplesToTurtle(toRdfJsQuads(webIdProfile)), {
+        headers: {
+          "Content-Type": "text/turtle",
+        },
+        url: MOCK_WEBID,
+      } as ResponseInit)
+    );
+
     await getProfileAll(MOCK_WEBID);
+    // The embedded fetch should have been used.
     expect(mockedFetcher.fetch).toHaveBeenCalled();
```

Comment:
I think here we actually need to test that we're using an unauthenticated fetcher for the first call, and then an authenticated fetcher for the rest, and actually assert that, or at least check that `mockedFetcher.fetch` has been called N number of times (2 if I'm reading this right)

Reply:
Should be fixed in fe6c5ea8f2b4419726d34b731398ab13e40a91cf